telemetry/vscode/src/generate.ts

```diff
@@ -132,7 +132,16 @@ const commonMetadata = [
   'requestId',
   'requestServiceType',
   'result',
-]
+] as const
+
+/** 
+ * While all of the commonMetadata fields are managed by the telemetry service,
+ * i.e. automatically computed, some fields make sense to also be specified by
+ * the caller.
+ *
+ * These values are allowed to be specified in .record() calls.
+ */
+const allowedMetadata: typeof commonMetadata[number][] = ['awsRegion']
```

Comment:
would `optionalMetadata` be a clearer name?

```suggestion
const optionalMetadata: typeof commonMetadata[number][] = ['awsRegion']
```